src/core/ext/filters/http/client/http_client_filter.cc

```diff
@@ -444,16 +444,26 @@ static grpc_error* init_call_elem(grpc_call_element* elem,
                                   const grpc_call_element_args* args) {
   call_data* calld = static_cast<call_data*>(elem->call_data);
   calld->call_combiner = args->call_combiner;
+  calld->recv_initial_metadata = nullptr;
```

Comment:
Some of these fields are never read until they are set in `hc_start_transport_stream_op_batch`, is it worth considering perf impact of leaving them uninitialized till then?

Same comment for all the cases where we can gaurantee a field will be set before it's read